eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Ppup.hs

```diff
@@ -91,8 +93,7 @@ data ShelleyPpupPredFailure era
     --  The first set contains the key hashes which were a part of the update.
     --  The second set contains the key hashes of the genesis keys.
     NonGenesisUpdatePPUP
-      !(Set (KeyHash 'Genesis (EraCrypto era)))
-      !(Set (KeyHash 'Genesis (EraCrypto era)))
+      !(Mismatch 'RelEQ (Set (KeyHash 'Genesis (EraCrypto era))))
```

Comment:
This is not an EQ relationship. This is a Subset relationship. I'd suggest we add it to the `Relation` type, since I suspect it will appear more than once.
```suggestion
      !(Mismatch 'RelSubset (Set (KeyHash 'Genesis (EraCrypto era))))
```

Reply:
💯 👍